src/sentry/tasks/integrations/github/open_pr_comment.py

```diff
@@ -102,3 +109,49 @@ def get_projects_and_filenames_from_source_file(
                 pr_filename.replace(code_mapping.source_root, code_mapping.stack_root)
             )
     return project_list, sentry_filenames
+
+
+def get_top_5_issues_by_count_for_file(
+    projects: Set[Project], sentry_filenames: Set[str]
+) -> list[dict[str, Any]]:
+    """Given a list of issue group ids, return a sublist of the top 5 ordered by event count"""
+    group_ids = list(
+        Group.objects.filter(
+            last_seen__gte=datetime.now() - timedelta(days=14),
+            project__in=projects,
+        ).values_list("id", flat=True)
+    )
+    project_ids = [p.id for p in projects]
+
+    request = SnubaRequest(
+        dataset=Dataset.Events.value,
+        app_id="default",
+        tenant_ids={"organization_id": projects[0].organization_id},
+        query=(
+            Query(Entity("events"))
+            .set_select(
+                [
+                    Column("group_id"),
+                    Function("count", [], "event_count"),
+                    Column("exception_frames.filename"),
+                ]
+            )
+            .set_groupby([Column("group_id"), Column("exception_frames.filename")])
+            .set_where(
+                [
+                    Condition(Column("project_id"), Op.IN, project_ids),
+                    Condition(Column("group_id"), Op.IN, group_ids),
+                    Condition(Column("timestamp"), Op.GTE, datetime.now() - timedelta(days=30)),
+                    Condition(Column("timestamp"), Op.LT, datetime.now()),
+                    Condition(
+                        Function("arrayElement", (Column("exception_frames.filename"), -1)),
+                        Op.IN,
+                        sentry_filenames,
```

Comment:
this looks at the filename at the top frame of a stack trace

Reply:
ideally this would be identical to how suspect commits work. currently we only look at the top frame of the stack trace

https://github.com/getsentry/sentry/blob/7e8008a7c2ab37663b3df8685b06381abbf23492/src/sentry/tasks/commit_context.py#L144-L146